### ISSUE TITLE
LAB1

### ISSUE THREAD

**Issue by bombaclaaaat** [bombaclaaaat/AI_LABY_20539#1]
import numpy as np
import matplotlib.pyplot as plt
import cv2
import tensorflow as tf
import tensorflow_datasets as tfds
print(tf.__version__)
mnist_dataset, mnist_info = tfds.load(name ='mnist', with_info=True, as_supervised=True);
mnist_train, mnist_test = mnist_dataset['train'], mnist_dataset['test']
num_valid_samples = 0.1 * mnist_info.splits['train'].num_examples
num_valid_samples = tf.cast(num_valid_samples, tf.int64)
num_test_samples = mnist_info.splits['test'].num_examples
num_test_samples = tf.cast(num_test_samples, tf.int64)
def scale(image, label):
    image = tf.cast(image, tf.float32)
    image /= 255.
    return image, label
scaled_train_and_validation_data = mnist_train.map(scale)
test_data = mnist_test.map(scale)

BUFSIZE = 10000
shuffled_train_and_validation_data = scaled_train_and_validation_data.shuffle(BUFSIZE)
valid_data = shuffled_train_and_validation_data.take(num_valid_samples)
train_data = shuffled_train_and_validation_data.skip(num_valid_samples)
test_data = test_data.batch(num_test_samples)
test_data = test_data.batch(num_test_samples)
valid_data = valid_data.batch(num_valid_samples)
train_data = train_data.batch(100)
valid_inputs, valid_targets = next(iter(valid_data))
print (valid_inputs.shape)
print (valid_targets.shape)
input_size = 784
output_size = 10
hidden_layer_size = 100

model = tf.keras.Sequential([
    tf.keras.layers.Flatten(input_shape=(28,28,1)), #warstwa wejść
    tf.keras.layers.Dense(hidden_layer_size, activation='relu'), #warstwa ukryta
    tf.keras.layers.Dense(hidden_layer_size, activation='relu'), #warstwa ukryta
    tf.keras.layers.Dense(output_size, activation='softmax') #warstwa wyjść
])
model.compile(optimizer='adam', loss='sparse_categorical_crossentropy', metrics=['accuracy'])
NUM_EPOCHS = 10
model.fit(train_data, epochs=NUM_EPOCHS, validation_data=(valid_inputs, valid_targets), verbose=2)
test_loss, test_accuracy = model.evaluate(test_data)
print('Test loss:{0:.2f}. Test accuracy:{1:.2f}%'.format(test_loss, test_accuracy*100))
